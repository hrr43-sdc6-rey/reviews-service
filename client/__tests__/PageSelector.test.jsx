import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16.1';
import PageSelector from '../components/PageSelector.jsx';

Enzyme.configure({ adapter: new Adapter() });

describe('PageSelector', () => {
  it('PageSelector snapshot should match shallow render with no pages of reviews', () => {
    const currPage = 1;
    const totalPages = 0;
    const jumpToPage = jest.fn();
    const wrapper = shallow(
      <PageSelector
        currPage={currPage}
        totalPages={totalPages}
        jumpToPage={jumpToPage}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('PageSelector snapshot should match shallow render with 1 page of reviews', () => {
    const currPage = 1;
    const totalPages = 1;
    const jumpToPage = jest.fn();
    const wrapper = shallow(
      <PageSelector
        currPage={currPage}
        totalPages={totalPages}
        jumpToPage={jumpToPage}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('PageSelector should render when on page 1 with 2 total pages', () => {
    const currPage = 1;
    const totalPages = 2;
    const jumpToPage = jest.fn();
    const wrapper = shallow(
      <PageSelector
        currPage={currPage}
        totalPages={totalPages}
        jumpToPage={jumpToPage}
      />,
    );
    expect(wrapper).toBeDefined();
  });

  it('PageSelector should render when on page 1 with 3 total pages', () => {
    const currPage = 1;
    const totalPages = 3;
    const jumpToPage = jest.fn();
    const wrapper = shallow(
      <PageSelector
        currPage={currPage}
        totalPages={totalPages}
        jumpToPage={jumpToPage}
      />,
    );
    expect(wrapper).toBeDefined();
  });

  it('PageSelector should render when on page 1 with 4 total pages', () => {
    const currPage = 1;
    const totalPages = 4;
    const jumpToPage = jest.fn();
    const wrapper = shallow(
      <PageSelector
        currPage={currPage}
        totalPages={totalPages}
        jumpToPage={jumpToPage}
      />,
    );
    expect(wrapper).toBeDefined();
  });

  it('PageSelector snapshot should match shallow render with current page of 1 and 8 total pages', () => {
    const currPage = 1;
    const totalPages = 8;
    const jumpToPage = jest.fn();
    const wrapper = shallow(
      <PageSelector
        currPage={currPage}
        totalPages={totalPages}
        jumpToPage={jumpToPage}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('PageSelector should render when on page 2 with 2 total pages', () => {
    const currPage = 2;
    const totalPages = 2;
    const jumpToPage = jest.fn();
    const wrapper = shallow(
      <PageSelector
        currPage={currPage}
        totalPages={totalPages}
        jumpToPage={jumpToPage}
      />,
    );
    expect(wrapper).toBeDefined();
  });

  it('PageSelector snapshot should match shallow render with current page of 2 and 3 total pages', () => {
    const currPage = 2;
    const totalPages = 3;
    const jumpToPage = jest.fn();
    const wrapper = shallow(
      <PageSelector
        currPage={currPage}
        totalPages={totalPages}
        jumpToPage={jumpToPage}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('PageSelector should render when on page 2 with 4 total pages', () => {
    const currPage = 2;
    const totalPages = 4;
    const jumpToPage = jest.fn();
    const wrapper = shallow(
      <PageSelector
        currPage={currPage}
        totalPages={totalPages}
        jumpToPage={jumpToPage}
      />,
    );
    expect(wrapper).toBeDefined();
  });

  it('PageSelector snapshot should match shallow render with current page of 2 and 5 total pages', () => {
    const currPage = 2;
    const totalPages = 5;
    const jumpToPage = jest.fn();
    const wrapper = shallow(
      <PageSelector
        currPage={currPage}
        totalPages={totalPages}
        jumpToPage={jumpToPage}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('PageSelector should render when on page 3 with 3 total pages', () => {
    const currPage = 3;
    const totalPages = 3;
    const jumpToPage = jest.fn();
    const wrapper = shallow(
      <PageSelector
        currPage={currPage}
        totalPages={totalPages}
        jumpToPage={jumpToPage}
      />,
    );
    expect(wrapper).toBeDefined();
  });

  it('PageSelector snapshot should match shallow render with current page of 3 and 4 total pages', () => {
    const currPage = 3;
    const totalPages = 4;
    const jumpToPage = jest.fn();
    const wrapper = shallow(
      <PageSelector
        currPage={currPage}
        totalPages={totalPages}
        jumpToPage={jumpToPage}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('PageSelector should render when on page 3 with 5 total pages', () => {
    const currPage = 3;
    const totalPages = 5;
    const jumpToPage = jest.fn();
    const wrapper = shallow(
      <PageSelector
        currPage={currPage}
        totalPages={totalPages}
        jumpToPage={jumpToPage}
      />,
    );
    expect(wrapper).toBeDefined();
  });

  it('PageSelector snapshot should match shallow render with current page of 3 and 8 total pages', () => {
    const currPage = 3;
    const totalPages = 8;
    const jumpToPage = jest.fn();
    const wrapper = shallow(
      <PageSelector
        currPage={currPage}
        totalPages={totalPages}
        jumpToPage={jumpToPage}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('PageSelector should render when on page 4 with 4 total pages', () => {
    const currPage = 4;
    const totalPages = 4;
    const jumpToPage = jest.fn();
    const wrapper = shallow(
      <PageSelector
        currPage={currPage}
        totalPages={totalPages}
        jumpToPage={jumpToPage}
      />,
    );
    expect(wrapper).toBeDefined();
  });

  it('PageSelector snapshot should match shallow render with current page of 4 and 5 total pages', () => {
    const currPage = 4;
    const totalPages = 5;
    const jumpToPage = jest.fn();
    const wrapper = shallow(
      <PageSelector
        currPage={currPage}
        totalPages={totalPages}
        jumpToPage={jumpToPage}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('PageSelector should render when on page 4 with 6 total pages', () => {
    const currPage = 4;
    const totalPages = 6;
    const jumpToPage = jest.fn();
    const wrapper = shallow(
      <PageSelector
        currPage={currPage}
        totalPages={totalPages}
        jumpToPage={jumpToPage}
      />,
    );
    expect(wrapper).toBeDefined();
  });

  it('PageSelector snapshot should match shallow render with current page of 4 and 8 total pages', () => {
    const currPage = 4;
    const totalPages = 8;
    const jumpToPage = jest.fn();
    const wrapper = shallow(
      <PageSelector
        currPage={currPage}
        totalPages={totalPages}
        jumpToPage={jumpToPage}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('PageSelector should render when on page 5 with 5 total pages', () => {
    const currPage = 5;
    const totalPages = 6;
    const jumpToPage = jest.fn();
    const wrapper = shallow(
      <PageSelector
        currPage={currPage}
        totalPages={totalPages}
        jumpToPage={jumpToPage}
      />,
    );
    expect(wrapper).toBeDefined();
  });

  it('PageSelector snapshot should match shallow render with current page of 5 and 6 total pages', () => {
    const currPage = 5;
    const totalPages = 6;
    const jumpToPage = jest.fn();
    const wrapper = shallow(
      <PageSelector
        currPage={currPage}
        totalPages={totalPages}
        jumpToPage={jumpToPage}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('PageSelector should render when on page 5 with 7 total pages', () => {
    const currPage = 5;
    const totalPages = 7;
    const jumpToPage = jest.fn();
    const wrapper = shallow(
      <PageSelector
        currPage={currPage}
        totalPages={totalPages}
        jumpToPage={jumpToPage}
      />,
    );
    expect(wrapper).toBeDefined();
  });

  it('PageSelector snapshot should match shallow render with current page of 5 and 8 total pages', () => {
    const currPage = 5;
    const totalPages = 8;
    const jumpToPage = jest.fn();
    const wrapper = shallow(
      <PageSelector
        currPage={currPage}
        totalPages={totalPages}
        jumpToPage={jumpToPage}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('PageSelector should render when on page 150 with 150 total pages', () => {
    const currPage = 150;
    const totalPages = 150;
    const jumpToPage = jest.fn();
    const wrapper = shallow(
      <PageSelector
        currPage={currPage}
        totalPages={totalPages}
        jumpToPage={jumpToPage}
      />,
    );
    expect(wrapper).toBeDefined();
  });

  it('PageSelector snapshot should match shallow render with current page of 201 and 202 total pages', () => {
    const currPage = 201;
    const totalPages = 202;
    const jumpToPage = jest.fn();
    const wrapper = shallow(
      <PageSelector
        currPage={currPage}
        totalPages={totalPages}
        jumpToPage={jumpToPage}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('PageSelector should render when on page 200 with 202 total pages', () => {
    const currPage = 200;
    const totalPages = 202;
    const jumpToPage = jest.fn();
    const wrapper = shallow(
      <PageSelector
        currPage={currPage}
        totalPages={totalPages}
        jumpToPage={jumpToPage}
      />,
    );
    expect(wrapper).toBeDefined();
  });

  it('PageSelector snapshot should match shallow render with current page of 453 and 456 total pages', () => {
    const currPage = 453;
    const totalPages = 456;
    const jumpToPage = jest.fn();
    const wrapper = shallow(
      <PageSelector
        currPage={currPage}
        totalPages={totalPages}
        jumpToPage={jumpToPage}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('PageSelector snapshot should match shallow render with current page of 30 and 120 total pages', () => {
    const currPage = 30;
    const totalPages = 120;
    const jumpToPage = jest.fn();
    const wrapper = shallow(
      <PageSelector
        currPage={currPage}
        totalPages={totalPages}
        jumpToPage={jumpToPage}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
