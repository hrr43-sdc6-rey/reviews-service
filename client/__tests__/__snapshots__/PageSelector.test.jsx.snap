// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageSelector PageSelector snapshot should match shallow render with 1 page of reviews 1`] = `""`;

exports[`PageSelector PageSelector snapshot should match shallow render with current page of 1 and 8 total pages 1`] = `
<div>
  <button
    className="currPage"
    type="button"
  >
    1
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    2
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    3
  </button>
  <button
    className="dots"
    type="button"
  >
    ...
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    8
  </button>
  <button
    className="next"
    onClick={[Function]}
    type="button"
  >
    &gt;
  </button>
</div>
`;

exports[`PageSelector PageSelector snapshot should match shallow render with current page of 2 and 3 total pages 1`] = `
<div>
  <button
    className="previous"
    onClick={[Function]}
    type="button"
  >
    &lt;
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    className="currPage"
    type="button"
  >
    2
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    3
  </button>
  <button
    className="next"
    onClick={[Function]}
    type="button"
  >
    &gt;
  </button>
</div>
`;

exports[`PageSelector PageSelector snapshot should match shallow render with current page of 2 and 5 total pages 1`] = `
<div>
  <button
    className="previous"
    onClick={[Function]}
    type="button"
  >
    &lt;
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    className="currPage"
    type="button"
  >
    2
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    3
  </button>
  <button
    className="dots"
    type="button"
  >
    ...
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    5
  </button>
  <button
    className="next"
    onClick={[Function]}
    type="button"
  >
    &gt;
  </button>
</div>
`;

exports[`PageSelector PageSelector snapshot should match shallow render with current page of 3 and 4 total pages 1`] = `
<div>
  <button
    className="previous"
    onClick={[Function]}
    type="button"
  >
    &lt;
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    2
  </button>
  <button
    className="currPage"
    type="button"
  >
    3
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    4
  </button>
  <button
    className="next"
    onClick={[Function]}
    type="button"
  >
    &gt;
  </button>
</div>
`;

exports[`PageSelector PageSelector snapshot should match shallow render with current page of 3 and 8 total pages 1`] = `
<div>
  <button
    className="previous"
    onClick={[Function]}
    type="button"
  >
    &lt;
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    2
  </button>
  <button
    className="currPage"
    type="button"
  >
    3
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    4
  </button>
  <button
    className="dots"
    type="button"
  >
    ...
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    8
  </button>
  <button
    className="next"
    onClick={[Function]}
    type="button"
  >
    &gt;
  </button>
</div>
`;

exports[`PageSelector PageSelector snapshot should match shallow render with current page of 4 and 5 total pages 1`] = `
<div>
  <button
    className="previous"
    onClick={[Function]}
    type="button"
  >
    &lt;
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    2
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    3
  </button>
  <button
    className="currPage"
    type="button"
  >
    4
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    5
  </button>
  <button
    className="next"
    onClick={[Function]}
    type="button"
  >
    &gt;
  </button>
</div>
`;

exports[`PageSelector PageSelector snapshot should match shallow render with current page of 4 and 8 total pages 1`] = `
<div>
  <button
    className="previous"
    onClick={[Function]}
    type="button"
  >
    &lt;
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    2
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    3
  </button>
  <button
    className="currPage"
    type="button"
  >
    4
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    5
  </button>
  <button
    className="dots"
    type="button"
  >
    ...
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    8
  </button>
  <button
    className="next"
    onClick={[Function]}
    type="button"
  >
    &gt;
  </button>
</div>
`;

exports[`PageSelector PageSelector snapshot should match shallow render with current page of 5 and 6 total pages 1`] = `
<div>
  <button
    className="previous"
    onClick={[Function]}
    type="button"
  >
    &lt;
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    className="dots"
    type="button"
  >
    ...
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    4
  </button>
  <button
    className="currPage"
    type="button"
  >
    5
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    6
  </button>
  <button
    className="next"
    onClick={[Function]}
    type="button"
  >
    &gt;
  </button>
</div>
`;

exports[`PageSelector PageSelector snapshot should match shallow render with current page of 5 and 8 total pages 1`] = `
<div>
  <button
    className="previous"
    onClick={[Function]}
    type="button"
  >
    &lt;
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    className="dots"
    type="button"
  >
    ...
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    4
  </button>
  <button
    className="currPage"
    type="button"
  >
    5
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    6
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    7
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    8
  </button>
  <button
    className="next"
    onClick={[Function]}
    type="button"
  >
    &gt;
  </button>
</div>
`;

exports[`PageSelector PageSelector snapshot should match shallow render with current page of 30 and 120 total pages 1`] = `
<div>
  <button
    className="previous"
    onClick={[Function]}
    type="button"
  >
    &lt;
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    className="dots"
    type="button"
  >
    ...
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    29
  </button>
  <button
    className="currPage"
    type="button"
  >
    30
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    31
  </button>
  <button
    className="dots"
    type="button"
  >
    ...
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    120
  </button>
  <button
    className="next"
    onClick={[Function]}
    type="button"
  >
    &gt;
  </button>
</div>
`;

exports[`PageSelector PageSelector snapshot should match shallow render with current page of 201 and 202 total pages 1`] = `
<div>
  <button
    className="previous"
    onClick={[Function]}
    type="button"
  >
    &lt;
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    className="dots"
    type="button"
  >
    ...
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    200
  </button>
  <button
    className="currPage"
    type="button"
  >
    201
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    202
  </button>
  <button
    className="next"
    onClick={[Function]}
    type="button"
  >
    &gt;
  </button>
</div>
`;

exports[`PageSelector PageSelector snapshot should match shallow render with current page of 453 and 456 total pages 1`] = `
<div>
  <button
    className="previous"
    onClick={[Function]}
    type="button"
  >
    &lt;
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    className="dots"
    type="button"
  >
    ...
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    452
  </button>
  <button
    className="currPage"
    type="button"
  >
    453
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    454
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    455
  </button>
  <button
    className="pages"
    onClick={[Function]}
    type="button"
  >
    456
  </button>
  <button
    className="next"
    onClick={[Function]}
    type="button"
  >
    &gt;
  </button>
</div>
`;

exports[`PageSelector PageSelector snapshot should match shallow render with no pages of reviews 1`] = `""`;
